great_expectations/data_context/store/datasource_store.py

```diff
@@ -62,20 +64,34 @@ def remove_key(self, key: Union[DataContextVariableKey, GXCloudIdentifier]) -> N
         """
         return self._store_backend.remove_key(key.to_tuple())
 
-    def serialize(self, value: DatasourceConfig) -> Union[str, dict, DatasourceConfig]:
+    def serialize(
+        self, value: DatasourceConfig | FluentDatasource
+    ) -> Union[str, dict, DatasourceConfig]:
         """
         See parent 'Store.serialize()' for more information
         """
+        if isinstance(
+            value, FluentDatasource
+        ):  # TODO: this is gross, do something else (or do it somewhere else)
+            return value._json_dict()
         return self._serializer.serialize(value)
```

Comment:
It seems natural to move this logic to the `serializer.serialize()` method, but the problem is that there's not an obvious way to move the `deserialization` away from this class as well.